fv3core/stencils/neg_adj3.py

```diff
@@ -299,70 +298,80 @@ def compute(qvapor, qliquid, qrain, qsnow, qice, qgraupel, qcld, pt, delp, delz,
         delz: Vertical thickness of atmosphere layers (in)
         peln: Logarithm of interface pressure (in)
     """
-    grid = spec.grid
 
-    shape_ij = qgraupel.shape[0:2]
-    sum1 = utils.make_storage_from_shape(
-        shape_ij, origin=(0, 0), cache_key="neg_adj3_sum1"
-    )
-    sum2 = utils.make_storage_from_shape(
-        shape_ij, origin=(0, 0), cache_key="neg_adj3_sum2"
-    )
-    upper_fix = utils.make_storage_from_shape(
-        qvapor.shape, origin=(0, 0, 0), cache_key="neg_adj3_upper_fix"
-    )
-    lower_fix = utils.make_storage_from_shape(
-        qvapor.shape, origin=(0, 0, 0), cache_key="neg_adj3_lower_fix"
-    )
-    if spec.namelist.check_negative:
-        raise NotImplementedError("Unimplemented namelist value check_negative=True")
-    if spec.namelist.hydrostatic:
-        d0_vap = constants.CP_VAP - constants.C_LIQ
-        raise NotImplementedError("Unimplemented namelist hydrostatic=True")
-    else:
-        d0_vap = constants.CV_VAP - constants.C_LIQ
-    lv00 = constants.HLV - d0_vap * constants.TICE
-    fix_neg_water(
-        pt,
-        delp,
-        delz,
+    def __init__(
+        self,
+        grid,
+        namelist,
+        qvapor,
+        qgraupel,
+    ):
+
+        shape_ij = qgraupel.shape[0:2]
+        self._sum1 = utils.make_storage_from_shape(shape_ij, origin=(0, 0))
+        self._sum2 = utils.make_storage_from_shape(shape_ij, origin=(0, 0))
+        self._upper_fix = utils.make_storage_from_shape(qvapor.shape, origin=(0, 0, 0))
+        self._lower_fix = utils.make_storage_from_shape(qvapor.shape, origin=(0, 0, 0))
```

Comment:
I don't think upper_fix and lower_fix are needed as storages outside of the stencil. I think they had been removed awhile ago, but I think slipped back in

Reply:
So you would make them a tmp ? They are accessed as a 3d field (accessing offset in z) wouldn't that be a slow tmp ?

Reply:
yeah let gt4py allocate them rather than us. I doubt there would be much performance impact from it. we've wanted to reduce the number of temporary fields we allocate in the python space, and how many fields need to be passed to stencils. The exception is our workarounds for regions, but I thought in general it was still true the less we can avoid declaring temporary storages, the better.

Reply:
Sounds good, done.